components/org.wso2.carbon.identity.login.resolver/org.wso2.carbon.identity.login.resolver.regex/src/main/java/org/wso2/carbon/identity/login/resolver/regex/RegexLoginResolver.java

```diff
@@ -0,0 +1,205 @@
+            }
+        } catch (UserStoreException e) {
+            log.error("An error occurred while resolving user.", e);
+        }
+        return resolvedUserResult;
+    }
+
+    /**
+     * Resolves a user from the given login identifier + login hint and then returns the resolved claim URI and the
+     * user details if a matching user exists.
+     *
+     * @param loginIdentifier   The user login identifier input which is provided by the user during login.
+     * @param allowedAttributes The list of claim URIs which are allowed by admin for resolving the user.
+     * @param tenantDomain      The tenant domain of the user.
+     * @param hint              The login attribute claim hint.
+     * @return ResolvedUserResult object with user details and resolved claim details.
+     */
+    @Override
+    public ResolvedUserResult resolveUser(String loginIdentifier, List<String> allowedAttributes, String tenantDomain,
+                                          String hint) {
+
+        log.warn("User resolver with hint is not yet implemented. Proceeding without the hint.");
+        return resolveUser(loginIdentifier, allowedAttributes, tenantDomain);
+    }
+
+    /**
+     * Authenticates with the user login identifier. Resolves the user by matching the login identifier and the user
+     * credentials.
+     *
+     * @param loginIdentifier   The user login identifier input which is provided by the user during login.
+     * @param allowedAttributes The list of claim URIs which are allowed by admin for resolving the user.
+     * @param credential        The credentials of the user.
+     * @param tenantDomain      The tenant domain of the user.
+     * @return An AuthenticationResult object with SUCCESS authentication status if the user credentials are correct
+     * otherwise returns an AuthenticationResult with FAIL authentication status.
+     */
+    @Override
+    public AuthenticationResult authenticateWithIdentifier(String loginIdentifier, List<String> allowedAttributes,
+                                                           Object credential, String tenantDomain) {
+
+        AuthenticationResult authenticationResult =
+                new AuthenticationResult(AuthenticationResult.AuthenticationStatus.FAIL);
+        ClaimManager claimManager;
+        try {
+            if (allowedAttributes == null) {
+                return authenticationResult;
+            }
+            UserRealm userRealm = UserResolverUtil.getUserRealm(tenantDomain);
+            UniqueIDUserStoreManager userStoreManager = UserResolverUtil.getUserStoreManager(tenantDomain);
+            claimManager = userRealm.getClaimManager();
+            for (String claimURI : allowedAttributes) {
+                Claim claim = claimManager.getClaim(claimURI);
+                if (claim == null) {
+                    continue;
+                }
+                String regex = claim.getRegEx();
```

Comment:
Refer same comment for line 70

Reply:
Please refer to the comment at https://github.com/wso2-extensions/identity-governance/pull/652#discussion_r1084917570.